core/src/subgraph/instance_manager.rs

```diff
@@ -736,8 +742,8 @@ where
             &block_ptr,
         ));
         let id = format!("{}-dynamic", Uuid::new_v4().to_simple());
-        let operations = entity.write_operations(id.as_ref());
-        block_state.entity_operations.extend(operations);
+        let operations = entity.write_entity_operations(id.as_ref());
+        block_state.entity_cache.borrow_mut().append(operations);
```

Comment:
Yes, it is annoying, but comes from the fact that dynamic data sources are both metadata and just 'normal' entities. In particular, they get reverted like normal entities, and therefore need to be stored with history. I feel that passing a `Vec<MetadataOperation>` into `transact_block_operations` would muddy the water since we usually execute `MetadataOperation` without recording history, but in this case we would have to.

Reply:
Alright, the current solution is a perfectly good compromise, we can keep it.